### PR TITLE
Support offset in DateTimeTrigger

### LINKED ISSUE #4195: Allow offset for "at the time specified in an item's state" trigger
I'd like to use the trigger "at the time specified in an item's state" for a rule, but be able to configure an offset. Then I can define "trigger one hour before sunset".

## Your Environment

* Version used: (e.g., openHAB and add-on versions) 4.1.2

### PR DESCRIPTION
**Pull request by jimtng**
Resolve #4195

This should be merged before #4272 (Rules DSL support)

### CODE REVIEW

**Comment by rkoshak** [comment]
Don't forget to open an issue/PR on MainUI to add support to the UI.

I'm excited to see this new feature. Thanks!

**Comment by florian-h05** [comment]
> Don't forget to open an issue/PR on MainUI to add support to the UI.

IIRC Main UI is using the config description provided through REST, so no need to update it.

File: bundles/org.openhab.core.automation/src/main/java/org/openhab/core/automation/internal/module/handler/DateTimeTriggerHandler.java [hunk 1]
```diff
@@ -71,6 +73,7 @@ public class DateTimeTriggerHandler extends BaseTriggerModuleHandler
     private final @Nullable EventFilter eventFilter;
     private String cronExpression = CronAdjuster.REBOOT;
     private Boolean timeOnly = false;
+    private Long offset = 0L;
```
Comment (jlaur) [inline bundles/org.openhab.core.automation/src/main/java/org/openhab/core/automation/internal/module/handler/DateTimeTriggerHandler.java:76]: Would a [Duration](https://docs.oracle.com/en/java/javase/17/docs/api/java.base/java/time/Duration.html) make sense here?

Comment (jimtng) [reply bundles/org.openhab.core.automation/src/main/java/org/openhab/core/automation/internal/module/handler/DateTimeTriggerHandler.java:76]: The Configuration / web / rulesdsl only supports limited data types: STRING, DECIMAL, etc. Besides, the granularity of cron is only down to seconds, whereas a Duration can represent milli/nano seconds.

**Review by holgerfriedrich — APPROVED** [review]
LGMT, thank you!

**Comment by jimtng** [comment]
@holgerfriedrich thanks for reviewing and merging. Would you mind starting a core + distro build? Waiting for it in order to run tests for https://github.com/openhab/openhab-jruby/pull/278

**Comment by holgerfriedrich** [comment]
@jimtng I don't think I have permission to do this. Let's ask @kaikreuzer, @wborn, @J-N-K to trigger a core build....

**Comment by wborn** [comment]
I have triggered a Core build!

**Comment by jimtng** [comment]
Thanks, @wborn! Will it automatically trigger a distro build upon a successful core build?

**Comment by wborn** [comment]
No but I also just clicked that button. :rocket:

**Comment by jimtng** [comment]
Thanks!

**Pull request merged**